src/python/dxpy/asset_builder.py

```diff
@@ -222,18 +222,19 @@ def build_asset(args):
 
         if not args.json:
             print("\nStarted job '" + str(job_id) + "' to build the asset bundle.\n", file=sys.stderr)
-            try:
-                subprocess.check_call(["dx", "watch", job_id])
-            except subprocess.CalledProcessError as e:
-                if e.returncode == 3:
-                    # Some kind of failure to build the asset. The reason
-                    # for the failure is probably self-evident from the
-                    # job log (and if it's not, the CalledProcessError
-                    # is not informative anyway), so just propagate the
-                    # return code without additional remarks.
-                    sys.exit(3)
-                else:
-                    raise e
+            if not args.skip_watch:
```

Comment:
To avoid this kind of double negative you can define the flag as:

```
add_argument('--no-watch', ..., action="store_false", dest="watch")
```

and then do:

```
if args.watch:
    # dx watch
```

Also, in most cases "negative" flags should be named "--no-FOO".